datafusion/sql/src/expr/identifier.rs

```diff
@@ -186,7 +186,22 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
                             let s = &ids[0..ids.len()];
                             // safe unwrap as s can never be empty or exceed the bounds
                             let (relation, column_name) = form_identifier(s).unwrap();
-                            Ok(Expr::Column(Column::new(relation, column_name)))
+                            // sanity check on column
+                            match schema
```

Comment:
I think we are trying to avoid matching on `Err` and converting to `Ok` because each `Err` results in at least one `String` allocation (for the error message)

Since this code is potentially invoked many times during planning the allocations can add up and slow down planning

Is there any way to use a method that doesn't return `Err` here?

Reply:
The result returned is not Err in most cases, we can ignore the additional cost here?